src/js/background/context-menu.js

```diff
@@ -0,0 +1,371 @@
+    await relayContextMenus.menus.create(staticMenuData.manageAliases);
+
+    // Generate upgrade menu item for non-premium users
+    const canUserUpgradeToPremium = await relayContextMenus.utils.getUserStatus.canUpgradeToPremium();
+    if (canUserUpgradeToPremium) {
+      await relayContextMenus.menus.create([staticMenuData.upgradeToPremiumSeperator, staticMenuData.upgradeToPremium]);
+    }
+
+    // Set listerners
+    browser.storage.onChanged.addListener(relayContextMenus.listeners.onLocalStorageChange);        
+
+    // Refresh menus
+    await browser.menus.refresh();
+
+    return Promise.resolve(1)
+
+  },
+  listeners: {
+    onFillInAddressWithAliasId: async (info, tab) => {
+      // Trim the context menu id to get the alias reference ID
+      const selectedAliasId = info.menuItemId.replace(
+        "fx-private-relay-use-existing-alias_",
+        ""
+      );
+
+      // Get stored alias data
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+
+      // Select the correct alias from the stored alias data
+      const selectedAliasObject = relayAddresses.filter((alias) => {
+        return alias.id === parseInt(selectedAliasId, 10);
+      });
+
+      browser.tabs.sendMessage(
+        tab.id,
+        {
+          type: "fillTargetWithRelayAddress",
+          targetElementId: info.targetElementId,
+          relayAddress: selectedAliasObject[0],
+        },
+        {
+          frameId: info.frameId,
+        }
+      );
+      
+    },
+    onLocalStorageChange: async (changes, area) => {
+      let changedItems = Object.keys(changes);
+      for (let item of changedItems) {
+        if (item === "relayAddresses") {
+          await relayContextMenus.init();
+        }
+
+        if (item === "apiToken" && changes[item].newValue === undefined) {
+          // User has logged out. Remove all menu items.
+          await browser.menus.removeAll();
+        }
+      }
+    },    
+    onMakeRelayAddressForTargetElement: async (info, tab) => {
+      const pageUrl = new URL(info.pageUrl);
+      const newRelayAddress = await makeRelayAddress(pageUrl.hostname);
+    
+      if (newRelayAddress.status === 402) {
+        browser.tabs.sendMessage(tab.id, {
+          type: "showMaxNumAliasesMessage",
+        });
+        return;
+      }
+    
+      await browser.tabs.sendMessage(
+        tab.id,
+        {
+          type: "fillTargetWithRelayAddress",
+          targetElementId: info.targetElementId,
+          relayAddress: newRelayAddress,
+        },
+        {
+          frameId: info.frameId,
+        }
+      );
+    }
+  },
+  menus: {
+    create: async (data, opts=null, aliases) => {     
+      // If multiple items need to be created: 
+      if (Array.isArray(data)) {
+        data.forEach(async (menu) => {
+          await browser.menus.create(menu, relayContextMenus.utils.onCreatedCallback);
+        });
+        
+        return Promise.resolve(1)
+      }
```

Comment:
I only see a single place where we call `relayContextMenus.menus.create` with an array argument. (i.e., `relayContextMenus.menus.create([staticMenuData.upgradeToPremiumSeperator, staticMenuData.upgradeToPremium])` ...

And this type-specific block completely changes the behavior of the function - e.g., if you pass an array argument, you can't make use of the `opts` or `aliases` argument values.

Can we either: 

1. remove this array-specific block and just call:
```js
relayContextMenus.menus.create(staticMenuData.upgradeToPremiumSeperator);
relayContextMenus.menus.create(staticMenuData.upgradeToPremium);
```

2. move this array-specific block to a new method:
```js
    createMultiple: async (data) => {     
      // If multiple items need to be created: 
      data.forEach(async (menu) => {
        await browser.menus.create(menu, relayContextMenus.utils.onCreatedCallback);
      });
      return Promise.resolve(1)
    }
```

Reply:
67d2c48

Reply:
Note – Vincent did the right solve — just make'em two calls.